app/models/current.rb

```diff
@@ -0,0 +1,18 @@
+class Current < ActiveSupport::CurrentAttributes
+  attribute :user
+
+  def account
+    return nil unless account_id.present?
+    @account ||= user&.accounts&.find(account_id)
+  end
+
+  private
+
+  def account_id
+    session[:account_id] ||= default_account&.id
```

Comment:
This felt like a bit of an unexpected side effect (assigning something to the `session`).

Also, out of curiosity, are there other areas where the `account_id` might get set or overridden?